packages/react-storage/src/components/StorageBrowser/createStorageBrowser/types.ts

```diff
@@ -319,7 +316,7 @@ export interface CreateStorageBrowserOutput<
   /**
    * @description action handler utility hook
    */
-  useAction: UseAction<DerivedActionHandlers<C>>;
+  useAction: UseAction<DerivedActionHandlers<NonNullable<C['custom']>>>;
```

Comment:
Couple things:
- Do we lose any type inference on overrides of `default` actions by leaving them out here?
- Can we move `NonNullable` inside `DerivedActionHandlers` for easier reading/understanding in the future?

Reply:
Confirmed the `useAction` hook does not support type inferrence from overwriting the default actions:

https://github.com/aws-amplify/amplify-ui/blob/e533dbbdade9d5ffdbd5aa7c446d958dd1980d43/packages/react-storage/src/components/StorageBrowser/useAction/types.ts#L50-L55